### PR TITLE
Fix playback confirmation

### PR DESCRIPTION
**Pull request by iwootten**
### What is the context of this PR?

Answers which have lists as their value are being incorrectly escaped. This causes problems when trying to use placeholder transformations for checkbox answers and each character of the list content is escaped.

### How to review 

Use the new test_placeholder_playback_list.json and select some toppings 🍕. Confirm they are correctly formatted on the confirmation page.

### CODE REVIEW

**Comment by iwootten** [comment]
[Fix playback confirmation feature](https://trello.com/c/nlmgzWE4/3487-fix-playback-confirmation-feature)

**Comment by ajmaddaford** [comment]
I think we should have a functional test to demonstrate the feature. Probably worth waiting for #2463 to merge first.

File: app/questionnaire/placeholder_parser.py [hunk 1]
```diff
@@ -40,9 +40,11 @@ def __call__(self, placeholder_list: Sequence[Mapping]) -> Mapping:
 
     def _lookup_answer(
         self, answer_id: str, list_item_id: str = None
-    ) -> Union[Markup, None]:
+    ) -> Union[Markup, Sequence[Markup], None]:
         answer = self._answer_store.get_answer(answer_id, list_item_id)
         if answer:
+            if isinstance(answer.value, list):
```
Comment (JamesGardiner) [inline app/questionnaire/placeholder_parser.py:46]: The use of isinstance is generally seen as a bad code smell or data choice. For example, in this example, if the value was always an iterable, the isinstance could be omitted. Instead we now have multiple return types for the placeholder_parser as well. This isn't a blocking issue for this PR, but it's worth mentioning as it is tech debt that could be refined.

**Review by JamesGardiner — APPROVED** [review]

**Comment by JamesGardiner** [comment]
Approved based on the functional test requested by @ajmaddaford being added

File: app/questionnaire/placeholder_parser.py (hunk 1 above)
Comment (iwootten) [reply app/questionnaire/placeholder_parser.py:46]: I'd considered the alternatives, but the fact is the value can be multiple types.

Refactoring the answerstore in this PR certainly seems overkill given that we frequently use isinstance elsewhere.

Comment (JamesGardiner) [reply app/questionnaire/placeholder_parser.py:46]: Yes, definitely overkill in this instance - are we tracking this kind of tech debt anywhere @ajmaddaford ?

**Comment by MebinAbraham** [comment]
👍

File: app/questionnaire/placeholder_parser.py (hunk 1 above)
Comment (ajmaddaford) [reply app/questionnaire/placeholder_parser.py:46]: Tech debt is tracked by raising a card with a 'Tech Debt' label. I don't think we have one for this so please add one. Ensure you add an explanation of why it's an issue and if possible a proposed fix.

**Review by ajmaddaford — APPROVED** [review]

**Pull request merged**